### PR TITLE
Support LayerWise for RTN/GPTQ

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

feature

## Description

usage
```python
from neural_compressor.torch.algorithms.layer_wise import load_empty_model
model = load_empty_model("hf-internal-testing/tiny-random-GPTJForCausalLM")

quant_config = GPTQConfig(
    use_layer_wise=True,
    model_path="hf-internal-testing/tiny-random-GPTJForCausalLM"
)
model = prepare(model, quant_config)
run_fn(model)
model = convert(model)

# RTN
from neural_compressor.torch.algorithms.layer_wise import load_empty_model

model = load_empty_model("hf-internal-testing/tiny-random-GPTJForCausalLM")
quant_config = RTNConfig(
    use_layer_wise=True, # auto-detect model_path 
)
model = prepare(model, quant_config)
model = convert(model)
```
client test optimization

- [x] clean weight: use to_empty() from torch
- [x] benchmark: taskset -c 0-11
- [x] don't frequently call gc: gc collect
- [x] packing accelerate: numaba.jit

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

Pre-CI

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/torch/algorithms/weight_only/modules.py [hunk 1]
```diff
@@ -300,7 +302,253 @@ def unpack_tensor_with_torch(self, packed_tensor):
                 accelerator.synchronize()
         return unpacked_tensor
 
-    def pack_tensor_with_numpy(self, raw_tensor):
+    @staticmethod
+    @numba.jit(nopython=True, parallel=True)
+    def pack_array_with_numba_b4_c32(
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/modules.py:307]: Please add UTs to verify the correctness of all pack functions.

**Review by yiliu30 — COMMENTED** [review]
Do we support non-transformer models? If yes, please add a unit test to demonstrate the usage.

**Review by yiliu30 — COMMENTED** [review]
Please simplify the import path of `load_empty_model` to `neural_compressor.torch.load_empty_model` for ease of use.

File: neural_compressor/torch/algorithms/weight_only/modules.py (hunk 1 above)
Comment (Kaihui-intel) [reply neural_compressor/torch/algorithms/weight_only/modules.py:307]: added https://github.com/intel/neural-compressor/pull/1883/commits/0ea77fd1ffdbf184abe7521a35f753079fb7bb54

**Review by n1ck-guo — APPROVED** [review]

File: neural_compressor/torch/algorithms/weight_only/rtn.py [hunk 2]
```diff
@@ -219,12 +238,17 @@ def convert(
...
+                m = m.to_empty(device=torch.device("meta"))
             if name == "":
                 return new_module
             else:
                 set_module(model, name, new_module)
             # Move modules back to the model device layer-by-layer
-            m.to(model_device)
-            new_module.to(model_device)
-        model.to(model_device)
+            if not use_layer_wise:
```
[the remark this answers is not in the document]
Comment (Kaihui-intel) [reply neural_compressor/torch/algorithms/weight_only/rtn.py:249]: Thanks, we will synchronize the layer-wise of auto-round next time

File: test/3x/torch/quantization/weight_only/test_gptq.py [hunk 3]
```diff
@@ -175,13 +175,24 @@ def test_act_order(self):
...
-    # TODO: (Xin) not implemented
+    def test_layer_wise(self):
+        model = copy.deepcopy(self.tiny_gptj)
+        quant_config = GPTQConfig()
+        model = prepare(model, quant_config)
+        run_fn(model)
+        model = convert(model)
+        q_label = model(self.example_inputs)[0]
+
+        from neural_compressor.torch.algorithms.layer_wise import load_empty_model
```
Comment (xin3he) [inline test/3x/torch/quantization/weight_only/test_gptq.py:186]: I agree with Yi that we should move this API to `neural_compressor.torch.utils`, others looks good to me.
Great PR.

**Review by xin3he — APPROVED** [review]

**Review by yiliu30 — APPROVED** [review]

**Pull request merged**